front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -42,7 +48,45 @@ const ManchetteWithSpaceTimeChartWrapper = ({
   );
 
   const [showSettingsPanel, setShowSettingsPanel] = useState(false);
-  const [settings, setSettings] = useState({ showConflicts: false });
+  const [settings, setSettings] = useState({
+    showConflicts: false,
+    showBAL: false,
+    showBAPR: false,
+    showTVM: false,
+    showSignalsStates: false,
+  });
+
+  const occupancyBlocks = projectPathTrainResult
+    .filter((train) => train.signal_updates)
+    .flatMap((train) =>
+      train.signal_updates.flatMap((block) => ({
+        timeStart: +new Date(train.departure_time) + block.time_start,
+        timeEnd: +new Date(train.departure_time) + block.time_end,
+        spaceStart: block.position_start,
+        spaceEnd: block.position_end,
+        aspect_label: block.aspect_label,
+        color:
+          block.aspect_label === 'VL' || block.aspect_label === '300VL'
+            ? OCCUPANCY_BLOCKS_COLORS.FREE
+            : block.aspect_label === 'S' ||
+                block.aspect_label === 'OCCUPIED' ||
+                block.aspect_label === 'C' ||
+                block.aspect_label === 'RRR'
+              ? OCCUPANCY_BLOCKS_COLORS.SEMAPHORE
+              : block.aspect_label === '(A)' || block.aspect_label === 'A'
+                ? OCCUPANCY_BLOCKS_COLORS.WARNING
+                : block.aspect_label === '300(VL)' ||
+                    block.aspect_label === '270A' ||
+                    block.aspect_label === '220A' ||
+                    block.aspect_label === '160A' ||
+                    block.aspect_label === '080A' ||
+                    block.aspect_label === '000'
+                  ? OCCUPANCY_BLOCKS_COLORS.GREY
+                  : '',
```

Comment:
I think this would be more readable with if/else blocks:

```ts
let color = '';
if (block.aspect_label === 'VL' || block.aspect_label === '300VL') {
    color = OCCUPANCY_BLOCKS_COLORS.FREE;
} else if (...) {
    ...
}
```

Reply:
Or even better, a `switch` or an object, since we're only testing a single value `aspect_label` here.

Reply:
Should we also extract it in a function to clear some space in the component ?

Reply:
Note: In the feature it's specified that the backend should returns a new field `aspect` which is a enum value.
Both `aspect_label` and `color` could be removed :shrug:

Reply:
I changed it, fixed

Reply:
Nice, much better!